harness/tests/test_ipc.py

```diff
@@ -237,20 +237,35 @@ def make_distributed_context(rank: int, cross_rank: int, local_rank: int) -> Any
 
 
 class TestPIDServer:
+    @staticmethod
+    def _worker_proc(
+        addr: Union[str, int, Tuple[str, int]],
+        keep_alive: bool = False,
+        sleep_time: float = 10,
+        repeat: int = 1,
+        crash: bool = False,
+    ) -> None:
+        with ipc.PIDClient(addr) as pid_client:
+            for _ in range(repeat):
+                if keep_alive:
+                    pid_client.keep_alive()
+                time.sleep(sleep_time)
+            if crash:
+                raise ValueError("Crashing...")
+
     def test_normal_execution(self) -> None:
         with ipc.PIDServer(addr=0, num_clients=2) as pid_server:
             assert pid_server.listener
             _, port = pid_server.listener.getsockname()
 
-            def worker_proc() -> None:
-                with ipc.PIDClient(port) as pid_client:
-                    for _ in range(5):
-                        pid_client.keep_alive()
-                        time.sleep(0.1)
-
+            # starting with python 3.8, an error is raised when process target is a local function
```

Comment:
Probably explain this in the git commit message, but not in a comment.  My thinking is that what you are saying applies to the transition from old code to new code more than it applies in particular to the new code.  If I saw this comment in the future and hadn't known what the test looked like before, I'd be pretty confused by it.